sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1101,9 +1116,9 @@ class Analyzer(override val catalogManager: CatalogManager)
       // The view's child should be a logical plan parsed from the `desc.viewText`, the variable
       // `viewText` should be defined, or else we throw an error on the generation of the View
       // operator.
-      case view @ View(desc, isTempView, child) if !child.resolved =>
+      case view @ View(Some(desc), isTempView, child) if !child.resolved =>
         // Resolve all the UnresolvedRelations and Views in the child.
-        val newChild = AnalysisContext.withAnalysisContext(desc) {
+        val newChild = AnalysisContext.withAnalysisContext(desc, isTempView) {
```

Comment:
I'm a bit confused here. For the new SQL temp view, we use `referredTempViewNames` to decide if we need to resolve temp views inside the view definition. What's wrong with it?

Reply:
yes, you are right. After `isReferredTempViewName` fix, this is no longer needed.